python/cudf/cudf/utils/ioutils.py

```diff
@@ -2099,3 +2095,96 @@ def _read_byte_ranges(
 
     for worker in workers:
         worker.join()
+
+
+def _get_remote_bytes_all(
+    remote_paths, fs, *, blocksize=_BYTES_PER_THREAD_DEFAULT
+):
+    if (
+        len(remote_paths) >= 8  # Heuristic to avoid fs.sizes
+        or max(sizes := fs.sizes(remote_paths)) <= blocksize
+    ):
+        # Don't bother braking up individual files
+        return fs.cat_ranges(remote_paths, None, None)
+    else:
+        # Construct list of paths, starts, and ends
+        paths, starts, ends = [], [], []
+        for i, remote_path in enumerate(remote_paths):
+            for j in range(0, sizes[i], blocksize):
+                paths.append(remote_path)
+                starts.append(j)
+                ends.append(min(j + blocksize, sizes[i]))
+
+        # Collect the byte ranges
+        chunks = fs.cat_ranges(paths, starts, ends)
+
+        # Construct local byte buffers
+        buffers = []
+        path_counts = np.unique(paths, return_counts=True)[1]
+        for i, remote_path in enumerate(remote_paths):
+            bytes_arr = bytearray()
+            for j in range(path_counts[i]):
+                bytes_arr.extend(chunks.pop(0))
+            buffers.append(bytes(bytes_arr))
+        return buffers
+
+
+def _get_remote_bytes_parquet(
+    remote_paths,
+    fs,
+    *,
+    columns=None,
+    row_groups=None,
+    blocksize=_BYTES_PER_THREAD_DEFAULT,
+):
+    if fsspec_parquet is None or (columns is None and row_groups is None):
+        return _get_remote_bytes_all(remote_paths, fs, blocksize=blocksize)
+
+    sizes = fs.sizes(remote_paths)
+    data = fsspec_parquet._get_parquet_byte_ranges(
```

Comment:
Hmm do we have to use an internal function? Are there no public APIs that would serve our purposes?

Reply:
For now, our options are:
- Use this "private" API (which I wrote a few years ago, and then never touched again...)
- Write our own version of the same logic in cudf
- Contribute an upstream PR to propose a public API for something similar}

My sense is that there is relatively little risk in using the "private" API for now, and that we can hopefully drop it once a long-term kvikio solution is available.

Reply:
A bit more context about the "public" APIs available in fsspec:

The "public" API is [open_parquet_file](https://github.com/fsspec/filesystem_spec/blame/7793ab89a56d70814de6d123c257e81f8e5bd52f/fsspec/parquet.py#L17), which opens a single parquet file with the "known" column and row-group ranges prefetched into an internal cache. For our use case, we just want the logic that fsspec uses to populate that internal cache.